File: Frontend/Web/src/app/api/api.config.ts
// API Configuration
export const API_CONFIG = {
    BASE_URL: 'http://localhost:8000',
    TIMEOUT: 30000,
    VERSION: 'v1'
};

// API Endpoints
export const ENDPOINTS = {
    // Auth
    AUTH: {
        REGISTER_STUDENT: '/auth/register/student',
        REGISTER_ENTERPRISE: '/auth/register/enterprise',
        LOGIN: '/auth/login',
    },

    // Students
    STUDENTS: {
        PROFILE: '/students/me',
        UPDATE_PROFILE: '/students/me/profile',
        UPLOAD_RESUME: '/students/me/resume',
        UPLOAD_PROFILE_PICTURE: '/students/me/profile-picture',
    },

    // Enterprises
    ENTERPRISES: {
        PROFILE: '/enterprises/me',
        UPDATE_PROFILE: '/enterprises/me/profile',
        UPLOAD_LOGO: '/enterprises/me/logo',
    },

    // PFE Listings
    PFE: {
        EXPLORE: '/api/pfe/explore',
        LISTINGS: '/api/pfe/listings',
    }
};

File: Frontend/mobile/lib/core/config/api_config.dart
class ApiConfig {
  // Base URL - Change this for different environments
  static const String baseUrl = 'http://localhost:8000';

  // Auth endpoints

  static const String login = '$baseUrl/auth/login';
  static const String logout = '$baseUrl/auth/logout';
  static const String refreshToken = '$baseUrl/auth/refresh';
  static const String googleAuth = '$baseUrl/auth/google';
  static const String forgotPassword = '$baseUrl/auth/forgot-password';
  static const String resetPassword = '$baseUrl/auth/reset-password';
  static const String verifyEmail = '$baseUrl/auth/verify-email';

  // Student endpoints
  static const String studentProfile = '$baseUrl/students/me';
  static const String updateProfile = '$baseUrl/students/me/profile';
  static const String uploadResume = '$baseUrl/students/me/resume';
  static const String uploadProfilePicture =
      '$baseUrl/students/me/profile-picture';
  static const String registerStudent = '$baseUrl/auth/register/student';

  // Enterprise endpoints
  static const String enterpriseProfile = '$baseUrl/enterprises/me';
  static const String updateEnterpriseProfile =
      '$baseUrl/enterprises/me/profile';
  static const String uploadEnterpriseLogo = '$baseUrl/enterprises/me/logo';
  static const String registerEnterprise = '$baseUrl/auth/register/enterprise';

  // Headers
  static Map<String, String> get defaultHeaders => {
    'Content-Type': 'application/json',
    'Accept': 'application/json',
  };

  static Map<String, String> authHeaders(String token) => {
    ...defaultHeaders,
    'Authorization': 'Bearer $token',
  };
}
